components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java

```diff
@@ -78,19 +107,104 @@ public Response apisCopyApiPost(String newVersion,String apiId)
     @Override
     public Response apisApiIdGet(String apiId,String accept,String ifNoneMatch,String ifModifiedSince)
     throws NotFoundException {
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        //validate API id (provider's tenant = carbon context.tenant domain)
+        String[] apiIdDetails = apiId.split("-");
+        String apiName = apiIdDetails[0];
+        String version = apiIdDetails[1];
+        String providerName = apiIdDetails[2];
+        String providerNameEmailReplaced = APIUtil.replaceEmailDomain(providerName);
+        boolean isTenantFlowStarted = false;
+        API apiToReturn = new API();
+        try {
+
+            APIIdentifier apiIdentifier = new APIIdentifier(providerNameEmailReplaced, apiName, version);
+            APIProvider apiProvider = APIManagerFactory.getInstance().getAPIProvider(providerName);
+            String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+
+            if (tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
```

Comment:
Please note that if the tenant domain becomes null, !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain) becomes true. Therefore it will start a tenant flow for null.
In order to guard this, tenantDomain != null is used.